test/test_regression/regressiontools.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import os.path
+import sys
+import argparse
+
+import numpy as np
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
+
+from neml import models, parse, drivers
+
+default_xml_file = "model.xml"
+default_model_name = "model"
+default_result_name = "data.csv"
+
+def generate_tensile_test(model, T = 300.0, erate = 1.0e-4, emax = 0.1,
+    nsteps = 100):
+  """
+    Generate results for a tension test
+  """
+  res = drivers.uniaxial_test(model, erate, T = T, emax = emax, 
+      nsteps = nsteps, full_results = True)
+  
+  data = np.concatenate((np.array(res['time'])[:,None], 
+    np.array(res['temperature'])[:,None],
+    np.array(res['strain']),
+    np.array(res['stress'])), axis = 1)
+
+  return data
+
+def run_compare_test(model, target_times, target_temps, target_strains):
+  """
+    Run through a single test
+  """
+  driver = drivers.Driver_sd(model)
+  for t,e,T in zip(target_times[1:], target_strains[1:], target_temps[1:]):
+    driver.strain_step(e,t,T)
+
+  data = np.concatenate((np.array(driver.t_int)[:,None], 
+    np.array(driver.T_int)[:,None],
+    np.array(driver.mechanical_strain_int),
+    np.array(driver.stress_int)), axis = 1)
+
+  return data
+
+def rtt(tdir):
+  model = parse.parse_xml(os.path.join(tdir, default_xml_file),
```

Comment:
can we add a description for this method

Reply:
Yep, it actually runs the test (added description)